python/ray/rllib/agents/impala/impala.py

```diff
@@ -113,6 +129,22 @@ def _init(self):
         if self.config["entropy_coeff"] < 0:
             raise DeprecationWarning("entropy_coeff must be >= 0")
 
+        if self.config["num_aggregation_workers"] > 0:
+            # Assign the pre-created aggregators to the optimizer
+            self.optimizer.aggregator.init(aggregators)
```

Comment:
Hm, why do you need to do this step separately? Why not have the AsyncSamplesOptimizer take in aggregators as part of its config?